contracts/StorageContract.sol

```diff
@@ -222,7 +231,7 @@ abstract contract StorageContract is DecentralizedKV {
 
         // Check if the data matches the hash in metadata and obtain the solution hash.
         bytes32 hash0 = keccak256(abi.encode(miner, bh, nonce));
-        hash0 = _verifySamples(shardId, hash0, miner, encodedSamples, proofs);
+        hash0 = _verifySamples(shardId, hash0, encodedSamples, inclusiveProofs, decodeProof);
```

Comment:
Does it make more sense to encode masks with decodeProof other than inclusiveProofs? Or as a separate parameter?